cypress/e2e/shared/notificationRules.test.ts

```diff
@@ -292,4 +302,457 @@ describe('NotificationRules', () => {
+          // 2. create check
+          const queryText = `from(bucket: \"${bucket.name}\")\n  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)\n  |> filter(fn: (r) => r[\"_measurement\"] == \"wumpus\")\n  |> filter(fn: (r) => r[\"_field\"] == \"mag\")`
+          cy.createCheck({
+            ...check,
+            orgID: org.id,
+            query: {
+              ...check.query,
+              text: queryText,
+              builderConfig: {
+                ...check.query.builderConfig,
+                buckets: [bucket.id],
+              },
+            },
+          }).then(resp => {
+            cy.wrap(resp.body).as('check')
+          })
+
+          cy.createEndpoint({
+            ...(endp as NotificationEndpoint),
+            orgID: org.id,
+          }).then(resp => {
+            cy.wrap(resp.body).as('endp')
+          })
+
+          cy.get<NotificationEndpoint>('@endp').then(endp => {
+            cy.createRule({
+              ...(rule as NotificationRule),
+              orgID: org.id,
+              endpointID: endp.id,
+            }).then(resp => {
+              cy.wrap(resp.body).as('rule')
+            })
+          })
+        })
+      })
+      cy.get<GenCheck>('@check').then(check => {
+        cy.get<NotificationEndpoint>('@endp').then(endp => {
+          cy.get<GenRule>('@rule').then(rule => {
+            writeMonitoringRecord(
+              {
+                check: check,
+                endp: endp,
+                rule: rule,
+                level: 'crit',
+                source: 'wumpus',
+                sent: 'true',
+                valField: {key: 'foo', val: 'bar'},
+                sourceTimestamp: calcNanoTimestamp('17m'),
+                statusTimestamp: calcNanoTimestamp('16m'),
+                message: 'This is a fake critical message',
+              },
+              '15m'
+            )
+
+            writeMonitoringRecord(
+              {
+                check: check,
+                endp: endp,
+                rule: rule,
+                level: 'info',
+                source: 'wumpus',
+                sent: 'false',
+                valField: {key: 'foo', val: 'bar'},
+                sourceTimestamp: calcNanoTimestamp('12m'),
+                statusTimestamp: calcNanoTimestamp('11m'),
+                message: 'This is a fake info message',
+              },
+              '10m'
+            )
+            writeMonitoringRecord(
+              {
+                check: check,
+                endp: endp,
+                rule: rule,
+                level: 'warn',
+                source: 'wumpus',
+                sent: 'true',
+                valField: {key: 'foo', val: 'bar'},
+                sourceTimestamp: calcNanoTimestamp('22m'),
+                statusTimestamp: calcNanoTimestamp('21m'),
+                message: 'This is a fake warning message',
+              },
+              '20m'
+            )
+            writeMonitoringRecord(
```

Comment:
this function is technically async because `cy.writeLPData` is async. So we need to chain off of it like `writeMonitoringRecord(...).then(() => ...`